### DISCUSSION TITLE
[Projects] Cannot prevent public projects

### DISCUSSION THREAD

**Discussion by ianwilliams1** [community/community#13190] [Other Feature Feedback, Questions, & Ideas]
(why is there no Feedback Category for **Projects**?)

Projects (both classic and beta) have only two options for Visibility.
**Public**: Anyone on the internet can see this project. You choose who can make changes.
**Private**: You choose who can see and make changes to this project.

User was actually expecting to provide Internal access but ended up giving Public in error.

At the Org level, under Projects (/organizations/orgname/settings/projects we only see:

> [ X ] Enable projects for the organization
> This allows members to create projects for the orgname organization. You can add issues from any orgname-owned repository. This does not affect projects in orgname-owned repositories in any way.
> 
> [ X ] Enable projects for all repositories
> This allows members to create projects in all repositories in the orgname organization. This affects every project at the repository level in the orgname organization. You can still disable projects for a specific repository in repository settings.
> 

We require a Global setting to disable the creation of "Public" projects. This would be consistent with Member privileges:
**Repository creation**
[  ] Public
Members will be able to create public repositories, visible to anyone.
**Repository outside collaborators**
[  ] Allow repository administrators to invite outside collaborators to repositories for this organization
    If disabled, only organization owners may invite collaborators to repositories
**Pages creation**
[  ] Public
    Members will be able to create public sites, visible to anyone
**Repository visibility change**
[  ]  Allow members to change repository visibilities for this organization
     If disabled, only organization owners can change repository visibilities.

Expectation:

- [ ] Global settings to PREVENT user from:

- Create a Public Project
- Convert an existing Project to Public

Expectation:

- [ ] Projects should have similar Visibility controls as Repos: [ Public | Internal | Private ].

**Comment by ianwilliams1** [comment]
Secondary to this ... 

It seems a new Project Base Role is set to "**Write**"
_Everyone in the organization can see and edit this project. Owners are admins of this project._

It would be my expectation the Base Role be "**Read**"
_Everyone in the organization can see this project._

And / Or similar to **Repos**, There should be a Base Permission for "Projects" and that be a default of "**Read**", or "No permission", or consistent with the Base Permission for repos.

**Comment by queenofcorgis** [comment]
Thanks for your feedback and explaining in detail your ideal workflow for this. While we don't have a category for Projects at this time, I did add the label so it can be properly sorted by the appropriate teams.
